spring-pulsar-docs/src/main/antora/modules/ROOT/pages/reference/pulsar/transactions.adoc

```diff
@@ -0,0 +1,222 @@
+
+Spring for Apache Pulsar provides the following:
+
+* `PulsarTransactionManager` - used with normal Spring transaction support (`@Transactional`, `TransactionTemplate`, etc)
+* Transactional `PulsarTemplate`
+* Transactional `@PulsarListener`
+* Transaction synchronization with other transaction managers
+
+NOTE: Transaction support has not been added to the Reactive components yet
+
+Transaction support is disabled by default.
+To enable support when using Spring Boot, simply set the `spring.pulsar.transaction.enabled` property.
+Further configuration options are outlined in each component section below.
+
+== Transactional Publishing with `PulsarTemplate`
+All send operations on a transactional `PulsarTemplate` look for an active transaction and enlist each send operation in the transaction (if one is found).
+
+=== Non-transactional use
+By default, a transactional `PulsarTemplate` can also be used for non-transactional operations.
+When an existing transaction is not found it will continue the send operation in a non-transactional fashion.
+However, if the template is configured to require transactions then any attempt to use the template outside the scope of a transaction results in an exception.
+
+TIP: A transaction can be started by a `TransactionTemplate`, a `@Transactional` method, calling `executeInTransaction`, or by a transactional listener container.
+
+=== Local Transactions
+We use the term "local" transaction to denote a Pulsar native transaction that is *not managed* by or associated with Spring's transaction management facility (i.e. `PulsarTransactionManager`).
+Conversely, a "synchronized" transaction is one that *is managed* by or associated with the `PulsarTransactionManager`.
+
+You can use the `PulsarTemplate` to execute a series of operations within a local transaction.
+The following example shows how to do so:
+
+[source, java]
+----
+var results = pulsarTemplate.executeInTransaction((template) -> {
+    var rv = new HashMap<String, MessageId>();
+    rv.put("msg1", template.send(topic, "msg1"));
+    rv.put("msg2", template.send(topic, "msg2"));
+    return rv;
+});
+----
+
+The argument in the callback is the template instance that the `executeInTransaction` method was invoked on.
+All operations on the template are enlisted in the current transaction.
+If the callback exits normally, the transaction is committed.
+If an exception is thrown, the transaction is rolled back.
+
+NOTE: If there is a synchronized transaction in process, it is ignored and a new "nested" transaction is used.
+
+=== Configuration
+The following transaction settings are available directly on the `PulsarTemplate` (via the `transactions` field):
+
+* `enabled` - whether the template supports transactions (default `false`)
+
+* `required` - whether the template requires transactions (default `false`)
+
+* `timeout` - duration of the transaction timeout (default `null`)
```

Comment:
Timeout in what?  i.e. seconds/millis/etc

Reply:
It is an actual `Duration` (i.e. implied in the passed in instance)